R/plot_scatter.R

```diff
@@ -0,0 +1,235 @@
+    net = c(0, 0, 0)
+  )
+
+  p <- ggplot2::ggplot(data, ggplot2::aes(x = .data$buildout, y = .data$phaseout, colour = .data$net)) +
+    ggplot2::geom_hline(yintercept = 0, colour = "#c0c0c0") +
+    ggplot2::geom_vline(xintercept = 0, colour = "#c0c0c0") +
+    ggplot2::geom_line(data = data_net_0) +
+    ggplot2::annotate(
+      geom = "text",
+      x = alignment_limit * 0.65,
+      y = -alignment_limit * 0.58,
+      label = "0% net deviation from scenario",
+      color = "white",
+      angle = -45,
+      size = 3
+    ) +
+    ggplot2::annotate(
+      geom = "text",
+      x = alignment_limit * 0.97,
+      y = alignment_limit * 0.87,
+      label = "Aligned buildout,\nAligned phaseout",
+      color = "#c0c0c0",
+      size = 3,
+      hjust = 1
+    ) +
+    ggplot2::annotate(
+      geom = "text",
+      x = -alignment_limit * 0.03,
+      y = alignment_limit * 0.87,
+      label = "Misaligned buildout,\nAligned phaseout",
+      color = "#c0c0c0",
+      size = 3,
+      hjust = 1
+    ) +
+    ggplot2::annotate(
+      geom = "text",
+      x = -alignment_limit * 0.97,
+      y = -alignment_limit * 0.87,
+      label = "Misaligned buildout,\nMisaligned phaseout",
+      color = "#c0c0c0",
+      size = 3,
+      hjust = 0
+    ) +
+    ggplot2::annotate(
+      geom = "text",
+      x = alignment_limit * 0.03,
+      y = -alignment_limit * 0.87,
+      label = "Aligned buildout,\nMisaligned phaseout",
+      color = "#c0c0c0",
+      size = 3,
+      hjust = 0
+    ) +
+    ggplot2::geom_point(ggplot2::aes(shape = .data$datapoint)) +
+    ggplot2::scale_x_continuous(
+      name = "Deviation from scenario value\nfor low-carbon technologies build-out",
+      labels = scales::percent,
+      limits = c(-alignment_limit, alignment_limit),
+      expand = ggplot2::expansion(mult = 0)
+    ) +
+    ggplot2::scale_y_continuous(
+      name = "Deviation from scenario value\nfor high-carbon technologies phase-out",
+      labels = scales::percent,
+      limits = c(-alignment_limit, alignment_limit),
+      expand = ggplot2::expansion(mult = 0)
+    ) +
+    ggplot2::scale_colour_gradient2(
+      name = "Net\ndeviation",
+      low = "#e10000",
+      mid = "white",
+      high = "#3d8c40",
+      midpoint = 0,
+      labels = scales::percent,
+      limits = c(-alignment_limit, alignment_limit),
+    ) +
+    ggplot2::scale_shape_manual(
+      name = "",
+      values = c("group" = 16, "benchmark" = 21, "company" = 16, "other" = 16),
+      labels = r2dii.plot::to_title
+    ) +
+    r2dii.plot::theme_2dii() +
+    ggplot2::theme(
+      panel.background = ggplot2::element_rect(fill = "#6c6c6c"),
+      legend.title = ggplot2::element_text(),
+      aspect.ratio = 1
+    ) +
+    ggplot2::labs(
+      title = title,
+      subtitle = subtitle,
+      caption = caption
+    )
+
+  if (data_level == "company") {
+    p <- p +
+      ggplot2::guides(
+        shape = "none"
+      )
+  }
+
+  p
+}
+# nolint end
```

Comment:
This makes me wonder if we are bringing `lintr` over as well

Reply:
let's not 😆